src/dialog-controller.ts

```diff
@@ -33,24 +53,64 @@ export class DialogController {
   /**
    * Creates an instance of DialogController.
    */
-  constructor(
-    renderer: Renderer,
-    settings: DialogSettings,
-    resolve: (data?: any) => void,
-    reject: (reason: any) => void) {
-    this.resolve = resolve;
-    this.reject = reject;
-    this.settings = settings;
+  constructor(renderer: Renderer, settings: DialogSettings) {
     this.renderer = renderer;
+    this.settings = settings;
   }
 
+  /**
+   * @internal
+   */
+  public initialize(controllerOrView: Controller | View): void {
+    if (isController(controllerOrView)) {
+      this.controller = controllerOrView;
+      this.view = this.controller.view;
+      return;
+    }
+    this.view = controllerOrView;
+  }
+
+  /**
+   * @internal
+   */
+  public deactivate(result: DialogCloseResult | DialogCloseError): Promise<void> {
+    if (this.controller) {
+      return invokeLifecycle(this.controller.viewModel, 'deactivate', result);
+    }
+    return Promise.resolve();
+  }
+
+  /**
+   * @internal
+   */
+  public show(): Promise<DialogOpenResult> {
+    return this.renderer.showDialog(this)
+      .then<DialogOpenResult>(() => {
+        return {
+          controller: this,
+          wasCancelled: false,
+          closeResult: new Promise((resolve, reject) => {
+            this.resolve = resolve;
+            this.reject = reject;
+          })
+        };
+      });
+  }
+
+  // /**
+  //  * @internal
+  //  */
+  // public hide(): Promise<void> {
```

Comment:
commented code

Reply:
Leftover, I decided not to do it in this PR.